From 10b3d38560ec53d980d13240559e6758c1113e21 Mon Sep 17 00:00:00 2001
From: Havard Graff <havard.graff@gmail.com>
Date: Thu, 21 Mar 2013 12:46:11 +0000
Subject: fs-rtp-discover-codecs: plug memoryleak

use g_list_delete_link to free the list as well
---
(limited to 'gst/fsrtpconference')

diff --git a/gst/fsrtpconference/fs-rtp-discover-codecs.c b/gst/fsrtpconference/fs-rtp-discover-codecs.c
index df316a4..c663225 100644
--- a/gst/fsrtpconference/fs-rtp-discover-codecs.c
+++ b/gst/fsrtpconference/fs-rtp-discover-codecs.c
@@ -536,7 +536,7 @@ remove_duplicates (GList *list)
       if (gst_caps_is_equal (codec_cap1->rtp_caps, codec_cap2->rtp_caps))
       {
         codec_cap_free (codec_cap2);
-        walk1 = g_list_remove_link (walk1, walk2);
+        walk1 = g_list_delete_link (walk1, walk2);
         goto again;
       }
     }
--
cgit v0.9.0.3
